extensions/binding/org.eclipse.smarthome.binding.mqttbroker/src/main/java/org/eclipse/smarthome/binding/mqttbroker/internal/NetworkUtils.java

```diff
@@ -0,0 +1,127 @@
+        } catch (SocketException ignored) {
+            // If we are not allowed to enumerate, we return an empty result set.
+            return Collections.enumeration(new ArrayList<NetworkInterface>());
+        }
+
+    }
+
+    /**
+     * Gets every IPv4 Address on each Interface except the loopback
+     * The Address format is ip/subnet
+     *
+     * @return The collected IPv4 Addresses
+     */
+    public Set<String> getInterfaceIPs() {
+        Set<String> interfaceIPs = new HashSet<String>();
+
+        // For each interface ...
+        for (Enumeration<NetworkInterface> en = getNetworkInterfaces(); en.hasMoreElements();) {
+            NetworkInterface networkInterface = en.nextElement();
+            boolean isLoopback = true;
+            try {
+                isLoopback = networkInterface.isLoopback();
+            } catch (SocketException ignored) {
+            }
+            if (!isLoopback) {
+                // .. and for each address ...
+                for (Iterator<InterfaceAddress> it = networkInterface.getInterfaceAddresses().iterator(); it
+                        .hasNext();) {
+
+                    // ... get IP and Subnet
+                    InterfaceAddress interfaceAddress = it.next();
+                    interfaceIPs.add(interfaceAddress.getAddress().getHostAddress() + "/"
+                            + interfaceAddress.getNetworkPrefixLength());
+                }
+            }
+        }
+
+        return interfaceIPs;
+    }
+
+    /**
+     * Get a set of all interface names.
+     *
+     * @return Set of interface names
+     */
+    public Set<String> getInterfaceNames() {
+        Set<String> result = new HashSet<String>();
+
+        try {
+            // For each interface ...
+            for (Enumeration<NetworkInterface> en = NetworkInterface.getNetworkInterfaces(); en.hasMoreElements();) {
+                NetworkInterface networkInterface = en.nextElement();
+                if (!networkInterface.isLoopback()) {
+                    result.add(networkInterface.getName());
+                }
+            }
+        } catch (SocketException ignored) {
+            // If we are not allowed to enumerate, we return an empty result set.
+        }
+
+        return result;
+    }
+
+    /**
+     * Determines every IP which can be assigned on all available interfaces
+     *
+     * @return Every single IP which can be assigned on the Networks the computer is connected to
+     */
+    public List<String> getNetworkIPs() {
+        return getNetworkIPs(getInterfaceIPs());
+    }
+
+    /**
+     * Takes the interfaceIPs and fetches every IP which can be assigned on their network
+     *
+     * @param networkIPs The IPs which are assigned to the Network Interfaces
+     * @return Every single IP which can be assigned on the Networks the computer is connected to
+     */
+    public List<String> getNetworkIPs(Set<String> interfaceIPs) {
```

Comment:
yes, my bad. I mixed methods and was refering to `getNetworkIPs(Set<String> interfaceIPs)` which is only used inside NetworkUtils.